### DISCUSSION TITLE
How to properly copy a APNG into a ImageTk.PhotoImage list (used for animations) while properly blending it with the previous frames

### DISCUSSION THREAD

**Discussion by maj113** [python-pillow/Pillow#8440] [Q&A] [answered]
Currently what happens is:

![image](https://github.com/user-attachments/assets/0295931e-28c7-4c23-9912-9130947956e2)

https://github.com/user-attachments/assets/025cdb38-49fd-469a-b70d-d53c08ec4d65

I use this to copy the frames:
```py
        ...
        raw_frames = []

        for i in range(self.frame_len):
            self.img.seek(i)
            raw_frames.append(self.img.copy())
        ...
 ```
 
The only other thing i do is thumbnail the frames and convert them to ImageTk.PhotoImages but the result is the same without thumbnailing
 
APNGs used: 

http://littlesvr.ca/apng/images/clock.png

http://littlesvr.ca/apng/images/blink.png

**Comment by radarhere** [comment, marked as answer]
Hi. Your problem isn't necessarily anything to do with ImageTk - Pillow is having a problem handling your image by itself.

You can see this by running
```python
from PIL import Image

with Image.open("blink.png") as im:
    im.seek(1)
    im.save("out.png")
```

I've created #8443 to address that.

**Reply by maj113** [reply]
Thank you for fixing this on such a short notice, I can confirm it fixes the issue on my end :)
